src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -220,6 +218,19 @@ public TextSubmission save(TextSubmission textSubmission) {
         return textSubmissions;
     }
 
+    /**
+     * Get a text submission of the given exercise that still needs to be assessed and lock the submission to prevent other tutors from receiving and assessing it.
+     *
+     * @param textExercise the exercise the submission should belong to
+     * @return a locked modeling submission that needs an assessment
+     */
+    public TextSubmission getLockedTextSubmissionWithoutResult(TextExercise textExercise) {
```

Comment:
```suggestion
    public TextSubmission findAndLockTextSubmissionToBeAssessed(TextExercise textExercise) {
```
or something similar, that indicates that this is not just a getter, but also performs a locking operation.

Reply:
Yes that is a good idea to rename it. It brings more meaning in this case